### DISCUSSION TITLE
composer update fails although the required php version is present

### DISCUSSION THREAD

**Discussion by kreatIL** [maennchen/ZipStream-PHP#215] [Q&A] [answered]
When I try to update via composer I get the following error:

- Installation request for maennchen/zipstream-php 2.2.1 -> satisfiable by maennchen/zipstream-php[2.2.1].
- maennchen/zipstream-php 2.2.1 requires php ^7.4 || ^8.0 -> your PHP version (7.3.9) does not satisfy that requirement.

PHP version is 7.4.29 (built: May 19 2022)
In my local development environment, running php 7.4.21, I do not get the error.

Any hint would be appreciated.

**Comment by maennchen** [comment, marked as answer]
Composer reports your PHP version as 7.3.9 and not 7.4.29.

Could it be that composer does not use the same binary as the one you call directly?

**Comment by kreatIL** [comment]
Thank you. This is kind of weird, because I thought composer would always run on the PHP version of the cli. I'll check again …

**Comment by kreatIL** [comment]
Yes, you're right. composer uses PHP 7.3.9 for whatever reason.
